include/zephyr/drivers/clock_control.h

```diff
@@ -221,7 +221,8 @@ static inline enum clock_control_status clock_control_get_status(const struct de
  * @param dev Pointer to the device structure for the clock controller driver
  *        instance
  * @param sys A pointer to an opaque data representing the sub-system
- * @param[out] rate Subsystem clock rate
+ * @param[out] rate Subsystem clock rate if clock is on, 0 otherwise
```

Comment:
Another option would be to return an error code (`-EAGAIN`). Opinions?

Reply:
Personally I'd vote for error code.

Reply:
what about: `@param[out] rate Valid clock rate on successful read else invalid value`
and something like:
```
@retval 0 successful rate reading
@retval -EAGAIN rate cannot be read, e.g. clock is off.
@retval -ENOTSUP feature is not supported.
@retval -ENODEV invalid device handle
```

Reply:
API meeting:

```
@retval 0 successful rate reading 
@retval -EAGAIN rate cannot be read. Some drivers do not support returning the rate when the clock is off. 
@retval -ENOTSUP feature is not supported. 
@retval -ENODEV invalid device handle
```

Reply:
I have adjusted the PR accordingly. Please review.